### ISSUE TITLE
Polling scenario in job

### ISSUE THREAD

**Issue by arrowcircle** [chaps-io/gush#98]
Hey! Is it possible to implement this:

* Check URL every minute until result is present or by global timeout
* Continue workflow when result is available

What is the proper way of doing such a thing with gush?

**Comment by pokonski** [comment]
Hello!

It's not strictly possible to build an explicit workflow that loops because we use acyclic graphs underneath, but you could make a blocking job that loops inside it waiting on data to appear.

**Comment by arrowcircle** [comment]
@pokonski Thanks for the answer.
My initial thought was to fail job if data is not ready, so it's restarted later. Will this work?

**Comment by pokonski** [comment]
That is right! Failed jobs do not enqueue further jobs of the workflow, so you can use it as a short circuit :) (Source: https://github.com/chaps-io/gush/blob/master/lib/gush/worker.rb#L23)

**Comment by arrowcircle** [comment]
Thanks, is there any proper way of failing a job and schedule it without raising error notifications system and with the compliance with gush API?

**Comment by pokonski** [comment]
That topic is outside of Gush, it raises an exception inside the job, you can rescue from it inside ActiveJob https://api.rubyonrails.org/classes/ActiveJob/Exceptions.html
